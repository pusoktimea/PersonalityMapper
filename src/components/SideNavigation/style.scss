@import '~compass-mixins/lib/compass';
@import "../../styles/_colors.scss";

$baseClass: '.activate-side-navigation';

#{$baseClass} {
  width: 180px;
  top: 66px;
  left: 0;
  z-index: 904;
  display: block;
  min-height: 100%;
  background-color: $mirage;
  height: 100%;
  position: fixed;
  color: $white;
  &_items {
    padding: 18px 0 0;
    overflow: hidden;
    width: auto;
    margin: 0;
    list-style: none;
    &_item {
      display: block;
      position: relative;
      float: none;
      padding: 0;
      a {
        height: 42px;
        padding-left: 18px;
        font-size: 13px;
        text-decoration: none;
        line-height: 42px;
        cursor: pointer;
        color: $white;
        overflow: hidden;
        text-shadow: none;
        white-space: nowrap;
        display: block;
        &:hover {
          color: $jungleGreen;
          border-right: 4px solid $jungleGreen;
          transition: all 0.5s ease;
          transition-property: top, height;
        }
        .activate-icon {
          margin-right: 18px;
          width: 16px;
          display: inline-block;
        }
      }
      &--selected {
        background-color: $jungleGreen !important;
        color: $white;
      }
    }
  }
}
