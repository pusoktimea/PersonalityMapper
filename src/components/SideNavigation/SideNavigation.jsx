import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import {Link} from 'react-router-dom';
import Icon from '../Icon';

import './style.scss';

const SideNavigation = ({isSideBarMinimised}) => (
  <div className={cx("activate-side-navigation", isSideBarMinimised && "activate-side-navigation--minimised")}>
    <ul className="activate-side-navigation_items">
      <li className="activate-side-navigation_items_item">
        <Link to="/tables"><Icon icon="home" />Tables</Link>
      </li>
      <li className="activate-side-navigation_items_item">
        <Link to="/search"><Icon icon="search" />Search</Link>
      </li>
      <li className="activate-side-navigation_items_item">
        <Link to="/reports"><Icon icon="pie-chart" />Reports</Link>
      </li>
      <li className="activate-side-navigation_items_item">
        <Link to="/users"><Icon icon="users" />Users</Link>
      </li>
      <li className="activate-side-navigation_items_item">
        <Link to="/roles"><Icon icon="calendar-check-o" />Roles</Link>
      </li>
    </ul>
  </div>
);

SideNavigation.propTypes = {
  isSideBarMinimised: PropTypes.bool
};

export default SideNavigation;
