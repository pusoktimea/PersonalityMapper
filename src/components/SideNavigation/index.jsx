import React from 'react';
import {Link} from 'react-router-dom';
import cookie from 'js-cookie';

import Icon from '../Icon';

import './style.scss';

const loggedInUser = cookie.get('loggedInUser');

const SideNavigation = () => (
  <div className="persmap-side-navigation">
    <ul className="persmap-side-navigation_items">
      <li className="persmap-side-navigation_items_item">
        <Link to="/"><Icon icon="users" />Dashboard</Link>
      </li>
      <li className="persmap-side-navigation_items_item">
        <Link to={`/profile/${loggedInUser}`}><Icon icon="calendar-check-o" />Profile</Link>
      </li>
    </ul>
  </div>
);

export default SideNavigation;
