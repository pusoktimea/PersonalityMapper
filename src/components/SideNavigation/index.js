import {connect} from 'react-redux';

import SideNavigation from './SideNavigation';

const mapStateToProps = (state) => {
  return {
    isSideBarMinimised: state.page.isSideBarMinimised
  };
};

export default connect(mapStateToProps)(SideNavigation);
