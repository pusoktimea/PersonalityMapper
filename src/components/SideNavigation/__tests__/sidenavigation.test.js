import 'raf/polyfill';

import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom';

import SideNavigation from '../SideNavigation';

describe('Side Navigation test', () => {
  it('renders and matches snapshot', () => {
    expect(
      renderer.create(
        <Router>
          <SideNavigation />
        </Router>
      ).toJSON()
    ).toMatchSnapshot();
  });
});
