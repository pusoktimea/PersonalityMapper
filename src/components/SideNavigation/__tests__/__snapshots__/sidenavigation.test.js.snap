// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side Navigation test renders and matches snapshot 1`] = `
<div
  className="activate-side-navigation"
>
  <ul
    className="activate-side-navigation_items"
  >
    <li
      className="activate-side-navigation_items_item"
    >
      <a
        href="/tables"
        onClick={[Function]}
      >
        <span
          className="activate-icon fa fa-home"
          id={undefined}
          onClick={undefined}
        />
        Tables
      </a>
    </li>
    <li
      className="activate-side-navigation_items_item"
    >
      <a
        href="/search"
        onClick={[Function]}
      >
        <span
          className="activate-icon fa fa-search"
          id={undefined}
          onClick={undefined}
        />
        Search
      </a>
    </li>
    <li
      className="activate-side-navigation_items_item"
    >
      <a
        href="/reports"
        onClick={[Function]}
      >
        <span
          className="activate-icon fa fa-pie-chart"
          id={undefined}
          onClick={undefined}
        />
        Reports
      </a>
    </li>
    <li
      className="activate-side-navigation_items_item"
    >
      <a
        href="/users"
        onClick={[Function]}
      >
        <span
          className="activate-icon fa fa-users"
          id={undefined}
          onClick={undefined}
        />
        Users
      </a>
    </li>
    <li
      className="activate-side-navigation_items_item"
    >
      <a
        href="/roles"
        onClick={[Function]}
      >
        <span
          className="activate-icon fa fa-calendar-check-o"
          id={undefined}
          onClick={undefined}
        />
        Roles
      </a>
    </li>
  </ul>
</div>
`;
